ilastik/applets/edgeTraining/opEdgeTraining.py

```diff
@@ -248,41 +255,66 @@ def setupOutputs(self):
-            if not feature_names:
-                # No features selected for this channel
-                continue
+            for c in range(self.VoxelData.meta.shape[-1]):
+                channel_name = self.VoxelData.meta.channel_names[c]
+                if channel_name not in channel_feature_names:
+                    continue
 
-            voxel_data = self.VoxelData[..., c : c + 1].wait()
-            voxel_data = vigra.taggedView(voxel_data, self.VoxelData.meta.axistags)
-            voxel_data = voxel_data[..., 0]  # drop channel
-            edge_features_df = rag.compute_features(voxel_data, feature_names)
+                feature_names = [decodeToStringIfBytes(f) for f in channel_feature_names[channel_name]]
+                if not feature_names:
+                    # No features selected for this channel
+                    continue
 
-            # if np.isnan(edge_features_df.values).any():
-            #    raise RuntimeError("Whoa, why are there NaN values in the feature matrix?")
+                voxel_data = self.VoxelData[..., c : c + 1].wait()
+                voxel_data = vigra.taggedView(voxel_data, self.VoxelData.meta.axistags)
+                voxel_data = voxel_data[..., 0]  # drop channel
+                edge_features_df = rag.compute_features(voxel_data, feature_names)
 
-            edge_features_df = edge_features_df.iloc[:, 2:]  # Discard columns [sp1, sp2]
+                # if np.isnan(edge_features_df.values).any():
+                #    raise RuntimeError("Whoa, why are there NaN values in the feature matrix?")
+
+                edge_features_df = edge_features_df.iloc[:, 2:]  # Discard columns [sp1, sp2]
 
-            # Prefix all column names with the channel name, to guarantee uniqueness
-            # (Generally a nice feature, but also required for serialization.)
-            edge_features_df.columns = [
-                channel_name + " " + feature_name for feature_name in edge_features_df.columns.values
-            ]
-            edge_feature_dfs.append(edge_features_df)
+                # Prefix all column names with the channel name, to guarantee uniqueness
+                # (Generally a nice feature, but also required for serialization.)
+                edge_features_df.columns = [
+                    channel_name + " " + feature_name for feature_name in edge_features_df.columns.values
+                ]
+                edge_feature_dfs.append(edge_features_df)
+
+            # Could use join() or merge() here, but we know the rows are already in the right order, and concat() should be faster.
+            all_edge_features_df = pd.DataFrame(rag.edge_ids, columns=["sp1", "sp2"])
+            all_edge_features_df = pd.concat([all_edge_features_df] + edge_feature_dfs, axis=1, copy=False)
+            result[0] = all_edge_features_df
+
+        else:
+            def normalize1(series):
+                series -= np.min(series)
+                series /= np.max(series)
```

Comment:
This will fail if `series` if of integer type, e.g.:
```
a = np.random.randint(0,255, (10,10))
a /= np.max(a)
```
gives:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: No loop matching the specified signature and casting
```
replace with `series = series / np.max(series)`